toolkit/scripts/toolchain.mk

```
@@ -169,10 +171,25 @@ $(final_toolchain): $(raw_toolchain) $(STATUS_FLAGS_DIR)/build_toolchain_srpms.f
 .SILENT: $(toolchain_rpms)
 
 ifeq ($(REBUILD_TOOLCHAIN),y)
+
+# Our manifest files should always track the contents of the freshly built archives exactly
+#   Currently non-blocking, to make this a blocking check switch to `print_error` instead of
+#   `print_warning`
+$(STATUS_FLAGS_DIR)/toolchain_verify.flag: $(TOOLCHAIN_MANIFEST) $(final_toolchain)
+	@echo Validating contents of toolchain against manifest...
+	tar -tzf $(final_toolchain) | sed 's+built_rpms_all/++g' | sed '/^$$/d' | sort > $(toolchain_actual_conents)
```

Suggestion to make it a bit shorter:

```suggestion
	tar -tzf $(final_toolchain) | grep -oP "[^/]+rpm$" | sort > $(toolchain_actual_conents)
```

---

sounds good (I borrowed it from the update_manifests.sh script, but I agree the grep is easier to understand)